pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTMethodDeclaration.java

```diff
@@ -124,6 +127,21 @@ public ASTNameList getThrows() {
     }
 
 
+    public List<ASTAnnotation> getDeclaredAnnotations() {
+
+        List<ASTAnnotation> result = new ArrayList<>();
+        ASTClassOrInterfaceBodyDeclaration decl = ((ASTClassOrInterfaceBodyDeclaration) jjtGetParent());
+        for (int i = 0; i < decl.jjtGetNumChildren() - 1; i++) {
+            if (!(decl.jjtGetChild(i) instanceof ASTAnnotation)) {
+                break;
```

Comment:
this assumes annotations will be the first children among modifiers, but that may be false. `public @Override static @PostConstruct void foo() { }` is completely valid Java code.

Reply:
We learn new things every day! But modifiers are not parsed into nodes, they just set a property on the method declaration, so I think Annotations are always the first children.
See the following, since Modifiers is a `#void`, so only annotation children are pushed on the tree
https://github.com/pmd/pmd/blob/d25fbcf8c0b52b5e97dca3ea169f0eed5ca53ecd/pmd-java/etc/grammar/Java.jjt#L1373-L1399

Reply:
You are absolutely right! Ignore my comment.

Reply:
You could use `decl.findChildrenOfType(ASTAnnotation.class)` - it should give you the same result. It however will loop through all children and doesn't stop on the first non-annotation node, loosing maybe some mini-performance.

Reply:
It's much more readable your way